arm9/src/unk_02006D98.c

```diff
@@ -0,0 +1,2488 @@
+    { 0x2C, 0x27 },
+    { 0x2D, 0x27 },
+    { 0x2E, 0x27 },
+    { 0x2F, 0x27 },
+    { 0x30, 0x27 },
+    { 0x28, 0x28 },
+    { 0x29, 0x28 },
+    { 0x2A, 0x28 },
+    { 0x2B, 0x28 },
+    { 0x2C, 0x28 },
+    { 0x2D, 0x28 },
+    { 0x2E, 0x28 },
+    { 0x2F, 0x28 },
+    { 0x30, 0x28 },
+    { 0x29, 0x29 },
+    { 0x2A, 0x29 },
+    { 0x2B, 0x29 },
+    { 0x2C, 0x29 },
+    { 0x2D, 0x29 },
+    { 0x2E, 0x29 },
+    { 0x2F, 0x29 },
+    { 0x2B, 0x2A },
+    { 0x2C, 0x2A },
+    { 0x2D, 0x2A },
+    { 0xFF, 0xFF },
+};
+
+const int UNK_020ECBD0[4][4] = {
+    { 160, 160, 224, 176 }, { 160, 160, 224, 176 }, { 160, 176, 224, 192 }, { 160, 192, 224, 208 }
+};
+
+const int UNK_020ECD4C[4][2][4] = { { { 0, 0, 0x50, 0x50 }, { 0x50, 0, 0xA0, 0x50 } },
+    {
+
+        { 0, 0x50, 0x50, 0xA0 }, { 0x50, 0x50, 0xA0, 0xA0 }
+
+    },
+    {
+
+        { 0, 0xA0, 0x50, 0xF0 }, { 0x50, 0xA0, 0xA0, 0xF0 }
+
+    },
+    {
+
+        { 0xA0, 0, 0xF0, 0x50 }, { 0xA0, 0x50, 0xF0, 0xA0 }
+
+    } };
+
+struct UnkStruct_02006D98_3 *UNK_02105AE8[] = {
+    UNK_020ECCE2,
+    UNK_020ECC78,
+    UNK_020ECC10,
+    UNK_020ECDCC,
+};
+
+extern void NNS_G2dSetupSoftwareSpriteCamera(void);
+extern void NNS_G3dGeFlushBuffer();
+extern void NNS_G2dDrawSpriteFast(s16 param0,
+    s16 param1,
+    int param2,
+    int param3,
+    int param4,
+    int param5,
+    int param6,
+    int param7,
+    int param8);
+
+THUMB_FUNC struct UnkStruct_02006D98 *FUN_02006D98(u32 heap_id)
+{
+    struct UnkStruct_02006D98 *ptr = AllocFromHeap(heap_id, sizeof(struct UnkStruct_02006D98));
+    ptr->unk298 = heap_id;
+    ptr->unk2E0 = 0;
+    ptr->unk29C = 0;
+    ptr->unk2A0 = 0x8000;
+    ptr->unk2A4 = 0;
+    ptr->unk2A8 = 0x80;
+
+    ptr->unk2AC = AllocFromHeap(heap_id, 0x8000);
+
+    ptr->unk2B0 = AllocFromHeap(heap_id, 0xc0);
+    MIi_CpuClearFast(0, ptr->unk2B0, 4);
+
+    ptr->unk2B4 = AllocFromHeap(heap_id, 0xc0);
+    MIi_CpuClearFast(0, ptr->unk2B4, 4);
+
+    for (int i = 0; i < 4; i++)
+    {
+        MIi_CpuClearFast(0, &ptr->unk000[i], sizeof(struct UnkStruct_02006D98_2));
+    }
+
+    NNS_G2dSetupSoftwareSpriteCamera();
+    ptr->unk2E3 = 0;
+
+    u8 *st8;
+    void *st4 = AllocAndReadWholeNarcMemberByIdPair(
+        NARC_POKETOOL_POKEGRA_OTHERPOKE, NARC_otherpoke_narc_0211_NCGR, ptr->unk298);
+    NNSG2dCharacterData *stc;
+    NNS_G2dGetUnpackedCharacterData(st4, &stc);
+
+    ptr->unk2B8.pixelFmt = stc->pixelFmt;
+    ptr->unk2B8.mapingType = stc->mapingType;
+    ptr->unk2B8.characterFmt = stc->characterFmt;
+    st8 = stc->pRawData;
+
+    FUN_02008A74(st8);
+    MI_CpuFill8(ptr->unk2AC, *st8, 0x8000);
+
+    for (int i = 0; i < 0x50; i++)
+    {
+        for (int j = 0; j < 0x28; j++)
+        {
+            ((u8 *)ptr->unk2AC)[0x5050 + i * 0x80 + j] = st8[i * 0x50 + j];
+        }
+    }
+
+    FreeToHeap(st4);
+    ptr->unk2E1 = 1;
+    ptr->unk2E2 = 1;
+
+    return ptr;
+}
+
+#ifdef NONMATCHING
+
+// mostly matching, only 2 register writes messed up
```

Comment:
probably sdk inlines are to blame

Reply:
What should we do in this case? Is it okay to copy them from the SDK? I see that some are already in the repo

Reply:
> Is it okay to copy them from the SDK?
Yes, just strip comments and sync the formatting/style to match the rest of the project.